Adding Ree6

DxsSucuk opened the pull request:
# Description

<!-- Please explain what is being changed or added as a short overview for this PR. Also, link existing relevant issues if they exist with resolves # -->
Adding a Egg for the Java Discord bot Ree6.

## Checklist for all submissions

<!-- insert X into the brackets to mark it as done (i.e. [x]). You can click preview to make the links appear clickable. -->

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [ ] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

<!-- If this is an egg update fill these out -->

* [x] You verify that the start command applied does not use a shell script
  * [x] If some script is needed then it is part of a current yolk or a PR to add one
* [x] The egg was exported from the panel

<!-- You can erase the new egg submission template if you're not adding a completely new egg -->

## New egg Submissions

1. [x] Does your submission pass tests (server is connectable)?
2. [ ] Does your egg use a custom docker image?
    * [ ] Have you tried to use a generic image?
    * [ ] Did you PR the necessary changes to make it work?
3. [x] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [x] Have you added a unique README.md for the egg you are adding according to the alphabetical order?
5. [x] You verify that the start command applied does not use a shell script
    * [x] If some script is needed then it is part of a current yolk or a PR to add one
6. [x] The egg was exported from the panel

QuintenQVD0 commented on bots/discord/ree6/egg-ree6.json:
```diff
@@ -0,0 +1,40 @@
+    "scripts": {
+        "installation": {
+            "script": "#!\/bin\/bash\n## this is a simple script to use the github API for release versions.\n## this requires the egg has a variable for GITHUB_PACKAGE and VERSION\n## this supports using oauth\/personal access tokens via GITHUB_USER and GITHUB_OAUTH_TOKEN (both are required.)\n## if you are getting hit with GitHub API limit issues then you need to have the user and token set.\n\ ...
+            "container": "debian:buster-slim",
```
old image, use debial bullseye-slim

QuintenQVD0 commented on bots/discord/ree6/egg-ree6.json:
```diff
@@ -0,0 +1,40 @@
+{
+    "_comment": "Pterodactyl Ree6 Egg ~ parkervcp, Software-Noob, Proxymiity, and Red-Thirten ~ 2021-11-29",
+    "meta": {
+        "version": "PTDL_v1",
```
this egg is old as it is still in v1 format

QuintenQVD0 commented on bots/discord/ree6/egg-ree6.json:
```diff
@@ -0,0 +1,40 @@
+    "file_denylist": [],
+    "startup": "java -Dnogui=true -jar Ree6.jar",
+    "config": {
+        "files": "{\r\n    \"config.yml\": {\r\n        \"parser\": \"file\",\r\n        \"find\": {# ################################\n# #                              #\n# # Ree6 Config File             #\n# # by Presti                    #\n# #                              #\n# ################################\n# \n\nconfig:\n  version: 2.1.0\n  creation: 1675970769277\nhikari:\n  sql:\n    us ...
```
the config is in .yml. so make variables and auto set them in the config file with the yaml parser

\+ this is not how the file parser works

DxsSucuk replied:
Shoud now use the yaml parser and only set the release token.

QuintenQVD0 commented on bots/discord/ree6/README.md:
```diff
@@ -0,0 +1,62 @@
+
+## Server Ports
+
+There are no ports required for Ree6.
```
A pterodactyl server will always have at least 1 port allocated even if the application doesn't use it

DxsSucuk replied:
Yeah but Ree6 itself does not require any ports.
A user still has to add one but its not needed by Ree6.

QuintenQVD0 commented:
and wy does this need a special egg? It looks like it can be ran fine with the generic java egg

DxsSucuk commented:
> and wy does this need a special egg? It looks like it can be ran fine with the generic java egg

Mainly because I got asked by users that use the Bot to make a way for them to make it easier hosting them via pterodactyl.

QuintenQVD0 commented:
> > and wy does this need a special egg? It looks like it can be ran fine with the generic java egg
> 
> Mainly because I got asked by users that use the Bot to make a way for them to make it easier hosting them via pterodactyl.

it is fine but some changes will need to be made as I pointed out before we can merge this

DxsSucuk commented:
> > > and wy does this need a special egg? It looks like it can be ran fine with the generic java egg
> > 
> > 
> > Mainly because I got asked by users that use the Bot to make a way for them to make it easier hosting them via pterodactyl.
> 
> it is fine but some changes will need to be made as I pointed out before we can merge this

Sadly this is actually my first really using the software and instead of reinventing the wheel as we would say I just took an existing config and modified it so its related to the current egg. And changed the installation script so it actually works with the project.

QuintenQVD0 commented:
> > > > and wy does this need a special egg? It looks like it can be ran fine with the generic java egg
> > > 
> > > 
> > > Mainly because I got asked by users that use the Bot to make a way for them to make it easier hosting them via pterodactyl.
> > 
> > 
> > it is fine but some changes will need to be made as I pointed out before we can merge this
> 
> Sadly this is actually my first really using the software and instead of reinventing the wheel as we would say I just took an existing config and modified it so its related to the current egg. And changed the installation script so it actually works with the project.

I will try to clean this up for you but not rn

DxsSucuk commented:
> > > > > and wy does this need a special egg? It looks like it can be ran fine with the generic java egg
> > > > 
> > > > 
> > > > Mainly because I got asked by users that use the Bot to make a way for them to make it easier hosting them via pterodactyl.
> > > 
> > > 
> > > it is fine but some changes will need to be made as I pointed out before we can merge this
> > 
> > 
> > Sadly this is actually my first really using the software and instead of reinventing the wheel as we would say I just took an existing config and modified it so its related to the current egg. And changed the installation script so it actually works with the project.
> 
> I will try to clean this up for you but not rn

No need to I would like to try it myself so I actually learn and improve upon it so when I gotta do this another time that I would understand it better. Thanks thou!

QuintenQVD0 commented:
I fix everything, post it here and then you can try it yourself and see how I have done it

QuintenQVD0 commented:
also a note, all changes must be made in a pterodactyl  panel and then exported. we don't allow you to edit the json file directly

QuintenQVD0 commented:
I started from 0 as your egg does not import
I made some basic and gave some examples with the config parser and fully fixt the install script so you don't need to edit that anymore. just the variables and the config parser.

```json
{
    "_comment": "DO NOT EDIT: FILE GENERATED AUTOMATICALLY BY PTERODACTYL PANEL - PTERODACTYL.IO",
    "meta": {
        "version": "PTDL_v2",
        "update_url": null
    },
    "exported_at": "2023-02-26T18:17:01+01:00",
    "name": "Ree6",
    "author": "josdekurk@gmail.com",
    "description": "All-in-one, open source and 100% free Discord Bot!",
    "features": null,
    "docker_images": {
        "Java_17": "ghcr.io\/pterodactyl\/yolks:java_17"
    },
    "file_denylist": [],
    "startup": "java -Dnogui=true -jar Ree6.jar",
    "config": {
        "files": "{\r\n    \"config.yml\": {\r\n        \"parser\": \"yaml\",\r\n        \"find\": {\r\n            \"dagpi.apitoken\": \"{{server.build.env.DAGPI_TOKEN}}\",\r\n            \"sentry.dsn\": \"{{server.build.env.SENTRY_DSN}}\"\r\n        }\r\n    }\r\n}",
        "startup": "{\r\n    \"done\": \"Finished Loading\"\r\n}",
        "logs": "{}",
        "stop": "^C"
    },
    "scripts": {
        "installation": {
            "script": "#!\/bin\/bash\r\n\r\napt update\r\napt install -y jq curl\r\n\r\n\r\nCONFIG_LINK=\"https:\/\/pteropaste.com\/0z244lpj9ku6\"\r\nMATCH=jar-with-dependencies.jar\r\n\r\n\r\n# get release info and download links\r\nLATEST_JSON=$(curl --silent \"https:\/\/api.github.com\/repos\/Ree6-Applications\/Ree6\/releases\/latest\")\r\nRELEASES=$(curl --silent \"https:\/\/api.github.com\/repos\/Ree6-Applications\/Ree6\/releases\")\r\n\r\nif [ -z \"${VERSION}\" ] || [ \"${VERSION}\" == \"latest\" ]; then\r\n    DOWNLOAD_URL=$(echo ${LATEST_JSON} | jq .assets | jq -r .[].browser_download_url | grep -i ${MATCH}| head -1)\r\nelse\r\n    VERSION_CHECK=$(echo ${RELEASES} | jq -r --arg VERSION \"${VERSION}\" '.[] | select(.tag_name==$VERSION) | .tag_name')\r\n    if [ \"${VERSION}\" == \"${VERSION_CHECK}\" ]; then\r\n        DOWNLOAD_URL=$(echo ${RELEASES} | jq -r --arg VERSION \"${VERSION}\" '.[] | select(.tag_name==$VERSION) | .assets[].browser_download_url' | grep -i ${MATCH} | head -1)\r\n    else\r\n        echo -e \"defaulting to latest release\"\r\n        DOWNLOAD_URL=$(echo ${LATEST_JSON} | jq .assets | jq -r .[].browser_download_url | grep -i ${MATCH}| head -1)\r\n    fi\r\nfi\r\n\r\n## Ree6 Setup\r\n[ ! -d \/mnt\/server ] && mkdir \/mnt\/server\r\ncd \/mnt\/server\r\necho -e \"\\nInstalling\/Updating Ree6...\\n\"\r\n\r\nif [ -f Ree6.jar ]; then\r\n    echo -e \"Updating Ree6.jar to latest version...\"\r\n    echo -e '\\t(Old version can be found at \"Ree6.jar-old\")'\r\n    mv -f Ree6.jar Ree6.jar-old\r\nfi\r\n\r\necho -e \"Running: curl -sSL -o Ree6.jar ${DOWNLOAD_LINK}\\n\"\r\ncurl -sSL -o Ree6.jar ${DOWNLOAD_LINK}\r\n\r\nif [ -f config.yml ]; then\r\n    echo -e \"Updating config.yml to latest version...\"\r\n    echo -e '\\t(Old version can be found at \"config.yml-old\")'\r\n    mv -f config.yml config.yml-old\r\nfi\r\necho -e \"Running: curl -sSL -o config.yml ${CONFIG_LINK}\\n\"\r\ncurl -sSL -o config.yml ${CONFIG_LINK}\r\n\r\necho -e \"\\nRee6 Successfully Installed!\"",
            "container": "debian:bullseye-slim",
            "entrypoint": "bash"
        }
    },
    "variables": [
        {
            "name": "[REQUIRED] Discord Bot Token",
            "description": "The token for your Discord bot. Learn how to obtain a token and configure your bot application properly here: https:\/\/jmusicbot.com\/getting-a-bot-token\/",
            "env_variable": "BOT_TOKEN",
            "default_value": "",
            "user_viewable": true,
            "user_editable": true,
            "rules": "required|string|",
            "field_type": "text"
        },
        {
            "name": "Dagpi token",
            "description": "https:\/\/dagpi.xyz\/",
            "env_variable": "DAGPI_TOKEN",
            "default_value": "",
            "user_viewable": true,
            "user_editable": true,
            "rules": "required|string",
            "field_type": "text"
        },
        {
            "name": "Sentry api token",
            "description": "",
            "env_variable": "SENTRY_DSN",
            "default_value": "",
            "user_viewable": true,
            "user_editable": true,
            "rules": "required|string",
            "field_type": "text"
        },
        {
            "name": "spotify client id",
            "description": "",
            "env_variable": "SPOT_CLIENT_ID",
            "default_value": "",
            "user_viewable": true,
            "user_editable": true,
            "rules": "required|string",
            "field_type": "text"
        },
        {
            "name": "spotify client secret",
            "description": "",
            "env_variable": "SPOT_CLIENT_SECRET",
            "default_value": "",
            "user_viewable": true,
            "user_editable": true,
            "rules": "required|string|",
            "field_type": "text"
        }
    ]
}
```

DxsSucuk commented:
How that does look quite more polished.
Issue with the installer script thou is that the actual file that users will need is always called Ree6-xxx-jar-with-dependencies and with the current looks it would just download the newest file.

QuintenQVD0 commented:
> How that does look quite more polished. Issue with the installer script thou is that the actual file that users will need is always called Ree6-xxx-jar-with-dependencies and with the current looks it would just download the newest file.

Yea you will need to add a version variable with the default value `latest` if uses want a other version then you can enter a other tag and it will download it

DxsSucuk commented:
Got around to actually install pterodactyl and set everything up.
It should now be fully ready to be reviewed and checked up to work.

DxsSucuk commented:
Anything else required for a merge? @QuintenQVD0 @parkervcp

parkervcp commented:
I can check this in a bit.

DxsSucuk commented:
> I can check this in a bit.

Thank you very much!

QuintenQVD0 commented:
> Anything else required for a merge? @QuintenQVD0 @parkervcp

Why did you do all that magic for the git release grab. we have a git release grab script in the repo

DxsSucuk commented:
> > Anything else required for a merge? @QuintenQVD0 @parkervcp
> 
> Why did you do all that magic for the git release grab. we have a git release grab script in the repo

WELL I found that out after I made it. And I decided to be too lazy. Also the modified script takes a specific JAR since Ree6 publishes 3 files. one without the dependencies, one with and another one is just a CLI Installer that help setup the config and download the latest file.

QuintenQVD0 commented:
> > > Anything else required for a merge? @QuintenQVD0 @parkervcp
> > 
> > 
> > Why did you do all that magic for the git release grab. we have a git release grab script in the repo
> 
> WELL I found that out after I made it. And I decided to be too lazy. Also the modified script takes a specific JAR since Ree6 publishes 3 files. one without the dependencies, one with and another one is just a CLI Installer that help setup the config and download the latest file.

I cleaned it up a litte. It now uses the offical release grab script, it checks the download url to see if it is valid. I moved the config download link to this repo as I will merge this. I removed the default values all the api fields and set the regex to nullable and an max of 128 characters. I indented the install script a bit. For the rest this is fine for me if it is for you

QuintenQVD0 commented:
@DxsSucuk Are all those api keys required or not? as it spits out a lot off erros if I don't have them

DxsSucuk commented:
> @DxsSucuk Are all those api keys required or not? as it spits out a lot off erros if I don't have them

They arent. The errors will still be shown but you can just ignore them.

QuintenQVD0 reviewed (approved):
LGTM! thank you
